### ISSUE TITLE
Create component "Filter by Volumes"

### ISSUE THREAD

**Issue by plutonik-a** [HistoryAtState/hsg-shell#293]
Display "Filter by volumes" in the result page. First 3 options are shown, the rest ist hidden and will be expanded by clicking the "show more" link. 

<img width="372" alt="bildschirmfoto 2017-11-13 um 14 16 11" src="https://user-images.githubusercontent.com/1157669/32727707-d528099c-c87d-11e7-8cad-a016caea9295.png">

Most volumes have long titles, that can hardly be displayed in a sidebar. 
See comment https://github.com/HistoryAtState/hsg-shell/issues/284#issuecomment-342542926
Awaiting an answer from @joewiz about the concept.

**Updates:** (from Joe in https://github.com/HistoryAtState/hsg-shell/issues/284#issuecomment-343969089)

* Replace dropdown arrows with +/- icons
* Text-color should be default text-color instead of link-color
* Display amount of hits next to the option
* "More Volumes" instead of "Show more"

> For volume titles, the "series" component you referred to should not be shown in the sidebar. Instead, we should just display the "sub-series", "volume-number", and "volume" titles. A sample query to retrieve these titles is:

```xquery
xquery version "3.1";

declare namespace tei="http://www.tei-c.org/ns/1.0";

declare function local:trim-words($string as xs:string, $number as xs:integer) {
    let $words := tokenize($string, "\s")
    return
        if (count($words) gt $number) then
            (
                subsequence($words, 1, ceiling($number div 2)) => string-join(" ")
                , "…"
                , $words[position() ge last() - floor($number div 2) + 1] => string-join(" ")
            )
            => string-join()
        else
            $string
};

element volumes {
    let $vols-in-db := collection("/db/apps/frus/volumes")/tei:TEI[.//tei:body/tei:div]/@xml:id
    let $vols := collection("/db/apps/frus/bibliography")/volume[@id = $vols-in-db]
    for $vol in $vols
    let $vol-id := $vol/@id
    let $title := 
        ($vol/title[@type eq "sub-series"], $vol/title[@type eq "volume-number"], $vol/title[@type eq "volume"])[. ne ""]
        => string-join(", ")
        => normalize-space()
        => local:trim-words(10)
    order by $vol-id
    return
        element volume { attribute id { $vol-id }, $title }
}
```

The first few results of this query are:

```xml
<volumes>
    <volume id="frus1865p4">1865, Part IV, Accompanying the…and Sympathy Inspired by These Events</volume>
    <volume id="frus1912">1912, With the Annual Message…Transmitted to Congress December 3, 1912</volume>
    <volume id="frus1913">1913, With the Address of…President to Congress December 2, 1913</volume>
    <!-- snip -->
</volumes>
```

---

Parent: #284 
Dependent on: #287

**Comment by joewiz** [comment]
Please see comments 2-7 from https://github.com/HistoryAtState/hsg-shell/issues/284#issuecomment-343969089.

**Comment by tuurma** [comment]
@plutonik-a is preparing a version with a truncated volume title as a temporary measure, awaiting decision how to present long titles for the sidebar

**Comment by plutonik-a** [comment]
**Status:**

The (multilined) volume titles are truncated to one line and their width depends on the width of the wrapping parent element. So in medium sized screens, the lines expand to full width, otherwise are limited in a sidebar. 
Nevertheless, the title beginnings are so similar, that the truncated lines do not differ, not even in medium sized screens.

<img width="317" alt="bildschirmfoto 2017-11-29 um 16 24 25" src="https://user-images.githubusercontent.com/1157669/33382910-492a0ec0-d522-11e7-84db-4fe2ab0a01ab.png">

**Comment by tuurma** [comment]
using @joewiz's suggestion for truncating the titles

![image](https://user-images.githubusercontent.com/449468/33395918-4977d1a8-d547-11e7-8544-e49c3c764deb.png)

**Comment by joewiz** [comment]
Thanks, Magda! That is our best option with the titles. The series is weird and inconsistent in its naming and numbering of volumes, and there’s only so much we can do.

**Comment by tuurma** [comment]
Thanks for confirmation @joewiz. As I see it we have potentially two more related things: 

1. limiting the number of volumes initially shown and making the rest available via `more volumes`
2. showing the number of results available for each filter option

Please, confirm if 1. is the desired solution, if so let's open a new issue and we will proceed with implementing it, though it will have low priority until we are done with core advanced search features.
2 is covered under #298 so let's move the discussion there.

**Comment by tuurma** [comment]
@joewiz confirms that limiting the number of volumes initially shown and making the rest available via 'more volumes` button is a good idea, I will follow up creating an issue
